backend/canisters/group/impl/src/queries/deleted_message.rs

```diff
@@ -0,0 +1,42 @@
+use crate::{read_state, RuntimeState};
+use group_canister::deleted_message::{Response::*, *};
+use ic_cdk_macros::query;
+
+#[query]
+fn deleted_message(args: Args) -> Response {
+    read_state(|state| deleted_message_impl(args, state))
+}
+
+fn deleted_message_impl(args: Args, runtime_state: &RuntimeState) -> Response {
+    let caller = runtime_state.env.caller();
+    let participant = match runtime_state.data.participants.get(caller) {
+        None => return CallerNotInGroup,
+        Some(p) => p,
+    };
+
+    if !participant.role.can_delete_messages(&runtime_state.data.permissions) {
```

Comment:
What about viewing your own message?
You may want to view it in case you deleted a few and can't remember which one was which.